takari-builder-common/src/test/java/io/takari/builder/internal/pathmatcher/PathNormalizerTest.java

```diff
@@ -0,0 +1,86 @@
+package io.takari.builder.internal.pathmatcher;
+
+import static io.takari.builder.internal.pathmatcher.PathNormalizer.normalize0;
+import static io.takari.builder.internal.pathmatcher.PathNormalizer.getCanonicalPath;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class PathNormalizerTest {
+
+  @Rule
+  public final TemporaryFolder temp = new TemporaryFolder();
+
+  @Test
+  public void testIsBasedirOrNestedFile() throws Exception {
+    Path basedir = temp.newFolder().getCanonicalFile().toPath();
+    PathNormalizer testee = PathNormalizer.create(basedir);
+
+    assertTrue(testee.isBasedirOrNestedFile(basedir.toString()));
+    assertTrue(testee.isBasedirOrNestedFile(basedir.resolve("nested").toString()));
+
+    assertFalse(testee.isBasedirOrNestedFile(temp.getRoot().getCanonicalPath()));
+  }
+
+  @Test
+  public void testAbsolutePath0() throws IOException {
+    String normalized = normalize0(temp.newFile().toPath());
+    assertTrue(normalized.charAt(0) == '/');
+    assertFalse(normalized.charAt(1) == '/'); // assert no doubles
+  }
+
+  @Test
+  public void testPathSeparator0() throws IOException {
+    assertFalse(normalize0(temp.newFile().toPath()).contains("\\"));
+  }
+
+  @Test
+  public void testIdempotency0() throws IOException {
+    String round1 = normalize0(temp.newFile().toPath());
+    assertEquals(round1, normalize0(round1));
+  }
+
+  @Test
+  public void testFileUrlPermission0() throws IOException {
+    // this is funny "\C:\path\to\file" path on Windows
+    String path = temp.newFile().toURI().toURL().openConnection().getPermission().getName();
+    String normalized = normalize0(path);
+    assertTrue(normalized.charAt(0) == '/');
```

Comment:
you could move these 3 assertions into a method: assertIsNormalized(String path)